### PR TITLE
Follow DRY principle and remove duplication

### PR DESCRIPTION
**Pull request by abhishekjain16**

### CODE REVIEW

**Comment by sgrif** [comment]
Thanks, but we don't accept cosmetic changes.

**Comment by abhishekjain16** [comment]
No problem :)

**Comment by rafaelfranca** [comment]
I think it is worth to merge. I'll review it.

File: actionview/lib/action_view/routing_url_for.rb [hunk 1]
```diff
@@ -147,5 +135,9 @@ def optimize_routes_generation? #:nodoc:
       def _generate_paths_by_default
         true
       end
+
+      def get_only_path(host)
```
Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: I think `only_path?` is better.

File: actionview/lib/action_view/routing_url_for.rb [hunk 2]
```diff
@@ -83,23 +83,11 @@ def url_for(options = nil)
...
       when Hash
         options = options.symbolize_keys
-        unless options.key?(:only_path)
-          if options[:host].nil?
-            options[:only_path] = _generate_paths_by_default
-          else
-            options[:only_path] = false
-          end
-        end
+        options[:only_path] = get_only_path(options[:host]) unless options.key?(:only_path)
```
Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: Please keep the `unless` in the beginning of the line

File: actionview/lib/action_view/routing_url_for.rb (hunk 1 above)
Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: Dislike `get_` prefixes. Rename to `only_path_for`

File: actionview/lib/action_view/routing_url_for.rb [hunk 3]
```diff
@@ -147,5 +135,9 @@ def optimize_routes_generation? #:nodoc:
       def _generate_paths_by_default
         true
       end
+
+      def get_only_path(host)
+        host.nil? ? _generate_paths_by_default : false
```
Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: Could save the method call and do `host ? false : generate_paths_by_default`.

Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: I think if host is false we don't want to return `false`

Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: @abhishekjain16 this is something you should check first though. Change to @kaspth's suggestion and see if any test fail.

Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: Good catch :+1:

File: actionview/lib/action_view/routing_url_for.rb (hunk 1 above)
Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: Rafaels suggestion was clearer :+1:

File: actionview/lib/action_view/routing_url_for.rb (hunk 3 above)
Comment (abhishekjain16) [inline actionview/lib/action_view/routing_url_for.rb]: @rafaelfranca @kaspth Great. 
I will change method name and will be checking for return value.

File: actionview/lib/action_view/routing_url_for.rb (hunk 2 above)
Comment (abhishekjain16) [inline actionview/lib/action_view/routing_url_for.rb]: @rafaelfranca I am not sure, is this the change you are looking for?

```ruby
unless options.key?(:only_path)
  options[:only_path] = get_only_path(options[:host])
end
```

File: actionview/lib/action_view/routing_url_for.rb [hunk 4]
```diff
@@ -147,5 +135,9 @@ def optimize_routes_generation? #:nodoc:
       def _generate_paths_by_default
         true
       end
+
+      def only_path_for(host)
```
Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: I still think `only_path?` is better :grin:

File: actionview/lib/action_view/routing_url_for.rb (hunk 2 above)
Comment (kaspth) [inline actionview/lib/action_view/routing_url_for.rb]: Yes, that's it 👍

File: actionview/lib/action_view/routing_url_for.rb (hunk 4 above)
Comment (abhishekjain16) [inline actionview/lib/action_view/routing_url_for.rb]: @kaspth Though this method name seems good considering it returns boolean value for now, but what if in future _generate_paths_by_default returns some other value than boolean?
What do you think?
Should we still go with only_path?

Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: if that happen we rename the method again. For now let's use `only_path?`

**Comment by abhishekjain16** [comment]
@rafaelfranca Done :+1:

File: actionview/lib/action_view/routing_url_for.rb [hunk 5]
```diff
@@ -83,23 +83,11 @@ def url_for(options = nil)
...
       when Hash
         options = options.symbolize_keys
-        unless options.key?(:only_path)
-          if options[:host].nil?
-            options[:only_path] = _generate_paths_by_default
-          else
-            options[:only_path] = false
-          end
-        end
+        options[:only_path] = only_path?(options[:host]) unless options.key?(:only_path)
```
Comment (rafaelfranca) [inline actionview/lib/action_view/routing_url_for.rb]: It is still missing the change of `unless` in the beginning of the line

**Comment by abhishekjain16** [comment]
@rafaelfranca Sorry. My bad. Change done now :+1:

**Pull request merged**